extra-modules/extra-pluralkit/src/main/kotlin/com/kotlindiscord/kord/extensions/modules/extra/pluralkit/PKExtension.kt

```diff
@@ -149,15 +148,21 @@ class PKExtension : Extension() {
                     return@action
                 }
 
-                val channel = topChannelFor(event)!!.asChannelOf<TopGuildMessageChannel>()
+                val channel = try {
+                    topChannelFor(event)?.asChannelOfOrNull<TopGuildMessageChannel>()
+                } catch (e: ClassCastException) {
```

Comment:
Isn't the point of `asChannelOfOrNull` to return `null` if the cast can't happen? I feel like this try-catch can never actually catch anything, though you should check what this function actually does before you take my word for it

Reply:
It's also worth leaving a comment to point out that this works around Kord's lack of support for forum channels - otherwise it shouldn't really be possible for the cast to fail; all channels that can have webhooks are top guild message channels

Reply:
> Isn't the point of `asChannelOfOrNull` to return `null` if the cast can't happen? I feel like this try-catch can never actually catch anything, though you should check what this function actually does before you take my word for it

Yes, however within the KDoc of the function it details that is will still throw if the channel is not of the specified type. I've not experimented enough to determine whether this is a copy-paste issue in the docs or factually correct.